test/e2e/upgrade_management_cluster_test.go

```diff
@@ -0,0 +1,352 @@
+	}, e2eConfig.GetIntervals(specName, "wait-controllers")...)
+
+	/*-------------------------------------------------*
+	| Pivote to run ironic/BMO and resources on target |
+	*--------------------------------------------------*/
+	By("Start pivoting")
+	By("Remove Ironic containers from the source cluster")
+	ephemeralCluster := os.Getenv("EPHEMERAL_CLUSTER")
+	if ephemeralCluster == KIND {
+		removeIronicContainers()
+	} else {
+		removeIronicDeployment()
+	}
+
+	By("Create Ironic namespace")
+	upgradeClusterClientSet := upgradeClusterProxy.GetClientSet()
+	ironicNamespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: os.Getenv("IRONIC_NAMESPACE"),
+		},
+	}
+	_, err = upgradeClusterClientSet.CoreV1().Namespaces().Create(ctx, ironicNamespace, metav1.CreateOptions{})
+	Expect(err).To(BeNil(), "Unable to create the Ironic namespace")
+
+	By("Configure Ironic Configmap")
+	configureIronicConfigmap(true)
+
+	By("Add labels to BMO CRDs")
+	labelBMOCRDs(nil)
+
+	By("Install BMO")
+	installIronicBMO(upgradeClusterProxy, "false", "true")
+
+	By("Install Ironic in the target cluster")
+	installIronicBMO(upgradeClusterProxy, "true", "false")
+
+	By("Restore original BMO configmap")
+	restoreBMOConfigmap()
+	By("Reinstate Ironic Configmap")
+	configureIronicConfigmap(false)
+
+	By("Add labels to BMO CRDs in the target cluster")
+	labelBMOCRDs(upgradeClusterProxy)
+
+	By("Ensure API servers are stable before doing move")
+	Consistently(func() error {
+		kubeSystem := &corev1.Namespace{}
+		return bootstrapClusterProxy.GetClient().Get(ctx, client.ObjectKey{Name: "kube-system"}, kubeSystem)
+	}, "5s", "100ms").Should(BeNil(), "Failed to assert bootstrap API server stability")
+	Consistently(func() error {
+		kubeSystem := &corev1.Namespace{}
+		return upgradeClusterClient.Get(ctx, client.ObjectKey{Name: "kube-system"}, kubeSystem)
+	}, "5s", "100ms").Should(BeNil(), "Failed to assert target API server stability")
+	By("Moving the cluster to self hosted")
+
+	cmd := exec.Command(clusterctlBinaryPath, "move", "--to-kubeconfig", upgradeClusterProxy.GetKubeconfigPath(), "-n", namespace, "-v", "10")
+	output, err := cmd.CombinedOutput()
+	Logf("move: %v", string(output))
+	Expect(err).To(BeNil())
+
+	By("Check if BMH is in provisioned state")
+	Eventually(func() error {
+		bmhList := &bmo.BareMetalHostList{}
+		if err := upgradeClusterClient.List(ctx, bmhList, client.InNamespace(namespace)); err != nil {
+			return err
+		}
+		for _, bmh := range bmhList.Items {
+			if !bmh.WasProvisioned() {
+				return errors.New("BMHs cannot be provisioned")
+			}
+		}
+		return nil
+	}, e2eConfig.GetIntervals(specName, "wait-object-provisioned")...).Should(BeNil())
+
+	Logf("Scale down KCP to 1 in order to 2 VMs for the workload cluster")
+	cmd = exec.Command("kubectl", "scale", "kcp", clusterName, "-n", namespace, "--replicas", "1", "--kubeconfig", upgradeClusterProxy.GetKubeconfigPath())
```

Comment:
Why should we not use the framework API here? It should be preference to use the native functions instead of using exec.Command

Reply:
for now we are using scaling since we cannot specify replicas from the e2e once we start using template parameterized from the e2e we can directly specify 1 kcp instead of starting with 3 then scale in to 1.
if #537 merged first we can rebase this PR else the template PR will consider using 1kcp and remove the scale part so I would prefer doing the change after merging one of these PRs.